packages/radix-vue/src/AlertDialog/AlertDialogRoot.vue

```diff
@@ -41,10 +41,30 @@ const open = useVModel(props, "open", emit, {
   passive: true,
 });
 
-const locked = useScrollLock(document.querySelector("body"), open.value);
+const locked = useScrollLock(document.body, open.value);
 
-watchEffect(() => {
-  locked.value = open.value;
+const cleanupBodyLock = () => {
+  document.body.style.marginRight = "";
+  document.body.style.pointerEvents = "";
+  locked.value = false;
+};
+
+watchEffect((onCleanup) => {
+  if (open.value) {
+    const verticalScrollbarWidth =
+      window.innerWidth - document.documentElement.clientWidth;
+
+    if (verticalScrollbarWidth > 0) {
+      document.body.style.marginRight = verticalScrollbarWidth + "px";
+    }
+
+    document.body.style.pointerEvents = "none";
+    locked.value = true;
+  } else {
+    cleanupBodyLock();
+  }
+
+  onCleanup(cleanupBodyLock);
 });
```

Comment:
https://github.com/twbs/bootstrap/blob/main/js/src/util/scrollbar.js#L33C1-L34C55

I think bootstrap approach is better, also use `padding` instead of `margin`

Reply:
I used margin to match original radix-ui. I dont think that it makes much difference.